### PR TITLE
Fix 3 failing tests, skip others, and bump `nmdc-schema` to `v11.0.0rc22`

### PR DESCRIPTION
**Pull request by aclum**
# Description

This PR fixes 3 failing tests:
test_endpoints.py
GOLD translator
test_ops.py (includes a nmdc-schema version bump)

This PR also removes -x from pytest to get all failed test.

Fixing the NEON translator tests and gold api test_ops are out of scope for this PR. @sujaypatil96  is working on those separately. This accounts for the remaining 6 failing tests.

Fixes # 
https://github.com/microbiomedata/nmdc-runtime/issues/675
https://github.com/microbiomedata/nmdc-runtime/issues/677
https://github.com/microbiomedata/nmdc-runtime/issues/679

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration, if it is not simply `make up-test && make test-run`.

- [ ] Test A
- [ ] Test B

**Configuration Details**: none

# Definition of Done (DoD) Checklist:
A simple checklist of necessary activities that add verifiable/demonstrable value to the product by asserting the quality of a feature, not the functionality of that feature; the latter should be stated as acceptance criteria in the issue this PR closes. Not all activities in the PR template will be applicable to each feature since the definition of done is intended to be a comprehensive checklist. Consciously decide the applicability of value-added activities on a feature-by-feature basis.

- [ ] My code follows the style guidelines of this project (have you run `black nmdc_runtime/`?)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (in `docs/` and in <https://github.com/microbiomedata/NMDC_documentation/>?)
- [ ] I have added tests that prove my fix is effective or that my feature works, incl. considering downstream usage (e.g. <https://github.com/microbiomedata/notebook_hackathons>) if applicable.
- [ ] New and existing unit and functional tests pass locally with my changes (`make up-test && make test-run`)

### CODE REVIEW

**Comment by sujaypatil96** [comment]
We need to all full module/file test skips for `tests/test_data/test_neon_benthic_data_translator.py` and `tests/test_data/test_neon_soil_data_translator.py` because we haven't made those two translators conformant with the berkeley schema as part of the initial rollout.

**Review by sujaypatil96 — COMMENTED** [review]
We need to add full module/file test skips for `tests/test_data/test_neon_benthic_data_translator.py` and `tests/test_data/test_neon_soil_data_translator.py` because we haven't made those two translators conformant with the berkeley schema as part of the initial rollout.

File: tests/test_data/test_gold_translator.py [hunk 1]
```diff
@@ -376,22 +407,25 @@ def test_get_lat_lon():
...
-            "seqMethod": [],
-        }
-    )
-    assert instrument is None
-
-    instrument = translator._get_instrument({"seqMethod": None})
-    assert instrument is None
+    with patch.object(
+        translator, "gold_nmdc_instrument_map_df", new=mock_gold_nmdc_instrument_map_df
+    ):
```
Comment (pkalita-lbl) [inline tests/test_data/test_gold_translator.py:412]: Is patching necessary here? Why not just pass `mock_gold_nmdc_instrument_map_df` to the `GoldStudyTranslator` constructor? That more accurately reflects how the class is used.

**Review by pkalita-lbl — COMMENTED** [review]
Overall seems fine to me. Minor quibble about how part of `test_gold_translator.py` is written.

File: tests/test_data/test_gold_translator.py (hunk 1 above)
Comment (aclum) [reply tests/test_data/test_gold_translator.py:412]: @sujaypatil96

**Comment by eecavanna** [comment]
I'm reviewing this PR now.

File: tests/test_data/test_gold_translator.py (hunk 1 above)
Comment (sujaypatil96) [reply tests/test_data/test_gold_translator.py]: Oh oops yes! much better than having to patch. Made a commit to do the above.

File: tests/test_api/test_endpoints.py [hunk 2]
```diff
@@ -295,7 +295,6 @@ def test_submit_workflow_activities(api_site_client):
                     "nmdc:dobj-11-bds7qq03",
                 ],
                 "type": "nmdc:ReadQcAnalysisActivity",
```
Comment (eecavanna) [inline tests/test_api/test_endpoints.py:297]: I'm wondering whether this `type` value has become `nmdc:ReadQcAnalysis` (i.e. no `Activity`) in the Berkeley schema.

**Comment by eecavanna** [comment]
Here's a summary—in my own words—of the changes present in this branch:

1. Updated `nmdc-schema` dependency from `11.0.0rc20` to `11.0.0rc22` and synchronized transitive dependencies (in `requirements/main.in` and `requirements/main.txt`)
2. Added missing `type` field to documents (in `tests/test_ops/test_ops.py`)
3. Updated the return type hint, and the docstring, of the `_get_instrument` function to reflect the fact the function returns an instrument _ID_ as opposed to an instrument _instance_ (in `nmdc_runtime/site/translation/gold_translator.py`). Also udated the calling code accordingly (in `tests/test_data/test_gold_translator.py`).
4. Removed the `-x` (stop on error) CLI option to `pytest` (in `nmdc_runtime/test.Dockerfile`)
5. Removed/replaced an obsolete field name in a test changesheet and document (in `tests/test_api/test_endpoints.py`)

Here's the `pytest` test results summary as of the latest commit on this branch:

image

**Review by eecavanna — APPROVED** [review]
Thanks, @aclum, for tackling this! Given that the 6 tests that still fail are ones that you expect to be failing, I'm OK with this being merged in as is.

**Comment by eecavanna** [comment]
> We need to add full module/file test skips for tests/test_data/test_neon_benthic_data_translator.py and tests/test_data/test_neon_soil_data_translator.py because we haven't made those two translators conformant with the berkeley schema as part of the initial rollout.

@sujaypatil96, I'll do that now in a follow-up commit, here on this branch.

**Comment by eecavanna** [comment]
As planned, the only tests that still fail are the four in `test_gold_api_ops.py`.

image

**Comment by eecavanna** [comment]
I skipped the final four failing tests in `test_gold_api_ops.py`. I added a `# FIXME` comment next to each `skip` decorator, like this:

```py
# FIXME: Update the API endpoint to work with the Berkeley schema, then un-skip this test.
#        Reference: https://docs.pytest.org/en/stable/how-to/skipping.html#skipping-test-functions
@pytest.mark.skip(
    reason="API endpoint has not been updated to work with the Berkeley schema yet"
)
```

I did something similar in the previous commit, when skipping the two NEON test modules.

I'm now waiting a few minutes for the GitHub Actions workflow to show that all the not-skipped tests are passing. Once that happens, I'll merge this into the `berkeley` branch, which will automatically update the Berkeley Runtime.

File: tests/test_api/test_endpoints.py (hunk 2 above)
Comment (eecavanna) [reply tests/test_api/test_endpoints.py:297]: Tagging @aclum about this for future reference. I'm about to merge this PR, which I think will take this thread further "out of sight, out of mind."

**Comment by eecavanna** [comment]
I think my adding "skip" annotations to some tests here is **inconsistent** with @aclum's vision for how things be; based on a conversation during today's metadata meeting. The previous 3 comments above list the tests I "skipped".

**Comment by eecavanna** [comment]
I skipped:
1. All tests in: `tests/test_data/test_neon_benthic_data_translator.py`
2. All tests in: `tests/test_data/test_neon_soil_data_translator.py`
3. All (4) tests in: `tests/test_ops/test_gold_api_ops.py`

**Comment by aclum** [comment]
tests/test_data/test_neon_benthic_data_translator.py is not part of MVP but @sujaypatil96  said he could have this done in a few more days 
tests/test_data/test_neon_soil_data_translator.py is not part of MVP but @sujaypatil96  said he could have this done in a few more days 
tests/test_ops/test_gold_api_ops.py I consider part of the MVP because it pertains to ingesting records from GOLD. This is the second most common ingest process after submission portal -> mongo.

**Comment by eecavanna** [comment]
Super helpful—thanks!

**Comment by eecavanna** [comment]
In that case, I'll **un-skip** _all_ of those tests via a PR shortly.

**Pull request merged**